### ISSUE TITLE
Support for condition-case-unless-debug

### ISSUE THREAD

**Issue by cpitclaudel** [jorgenschaefer/emacs-buttercup#45]
Hi Jorgen,

What's the right way to write a test to ensure that a function properly swallows errors? For example, this fails:

```
(describe "condition-case-unless-debug"
  (it "does not throw too much"
    (expect (condition-case-unless-debug nil (error "!!") (error nil)) :to-equal nil)))
```

I'd like to make sure that the function in fact does not throw errors; is there a way?

**Comment by jorgenschaefer** [comment]
Hello! The right way to do this right now is to use `(expect (lambda () (error "!!")) :to-throw 'error)` – does that answer your question?

**Comment by cpitclaudel** [comment]
No, not really :) I wondering how to make that code not throw; the exception should be swallowed by `condition-case-unless-debug`, but it seems that the environment in which buttercup runs that test disables `condition-case-unless-debug`; that's good in general, but not in this particular case.

**Comment by jorgenschaefer** [comment]
Oh, sorry, totally misunderstood the problem.

Yes, buttercup binds `debug-on-error`, because that's the only way buttercup can actually get a backtrace (Emacs Lisp limitation). You can `let`-bind `debug-on-error` yourself, of course, but do that at your own risk :-)

**Comment by cpitclaudel** [comment]
Ah, I see :) Makes sense! Perhaps this could be in the docs :)

**Comment by jorgenschaefer** [comment]
Good idea!
